src/antlr/QueryVisitor.ts

```diff
@@ -0,0 +1,412 @@
+import { Column, Lineage } from "../Lineage";
+import { SqlBaseVisitor } from "./SqlBaseVisitor";
+import { AbstractParseTreeVisitor } from "antlr4ts/tree/AbstractParseTreeVisitor";
+import { Range } from "../utils/getRange";
+import { ColumnRef, QuotableIdentifier, Relation } from "./model";
+import { ParserRuleContext } from "antlr4ts";
+import {
+  AliasedQueryContext,
+  ColumnReferenceContext,
+  DereferenceContext,
+  ExpressionContext,
+  FromClauseContext,
+  IdentifierContext,
+  NamedExpressionContext,
+  PredicatedContext,
+  PrimaryExpressionContext,
+  QueryContext,
+  QuotedIdentifierAlternativeContext,
+  RegularQuerySpecificationContext,
+  StarContext,
+  StrictIdentifierContext,
+  TableNameContext,
+  ValueExpressionDefaultContext
+} from "./SqlBaseParser";
+import { LineageContext } from "./LineageContext";
+
+export class QueryVisitor<TableData extends { id: string }, ColumnData extends { id: string }>
+  extends AbstractParseTreeVisitor<Lineage<TableData, ColumnData> | undefined>
+  implements SqlBaseVisitor<Lineage<TableData, ColumnData> | undefined> {
+  private id;
+
+  private topQuery = true;
+
+  // sequence generator for columns in this context
+  private columnIdSeq = 0;
+
+  // columns for this query extracted from SELECT
+  columns: Array<Column<ColumnData>> = [];
+
+  // relations for this context extracted from FROM
+  relations: Map<string, Relation<TableData, ColumnData>> = new Map();
+
+  isStar = false;
```

Comment:
As you see here: https://github.com/contiamo/rhombic/blob/89e8b798180a273934b19619bfc54cee46ca8efb/src/antlr/SqlBase.g4#L835 the lexem is asterisk but grammar rule name is "star". I'm following the grammar here to avoid confusion.

Reply:
Interesting ^^